src/components/studies/adherenceHooks.ts

```diff
@@ -88,6 +91,34 @@ export const useAdherenceAlerts = (
   )
 }
 
+export const useUpdateAdherenceAlerts = () => {
+  const {token} = useUserSessionDataState()
+  const queryClient = useQueryClient()
+
+  // update alerts status on the server
+  const update = async (props: {
+    studyId: string, 
+    alertIds: string[], 
+    action: 'READ' | 'UNREAD' | 'DELETE',
+  }): Promise<any> => {
+    return await AdherenceService.updateAdherenceAlerts(props.studyId, props.alertIds, props.action, token!)
+  }
+
+  // invalidate alerts data
+  const mutation = useMutation(update, {
+    onMutate: async props => {
+      queryClient.cancelQueries(ADHERENCE_KEYS.alertsAll(props.studyId))
+    },
+    onError: (err, variables, context) => {
+      console.log(err, variables, context)
+    }, 
+    onSettled: async (data, error, props) => {
+      queryClient.invalidateQueries({queryKey: ADHERENCE_KEYS.alertsAll(props.studyId), refetchActive: true, refetchInactive: true})
```

Comment:
I would think TypeScript would catch this if it was a problem, but this looks like the react-query v4 signature when you're using react-query v3.

See [QueryClient.invalidateQueries ](https://react-query-v3.tanstack.com/reference/QueryClient#queryclientinvalidatequeries)

Maybe after this change you could remove these extra parameters?

Also, if you want the cache data to be dropped (so `useAdherenceAlerts` would immediately enter the `loading` state, assuming there isn't a default initial data value), you'd want to use [resetQueries](https://react-query-v3.tanstack.com/reference/QueryClient#queryclientresetqueries)

```suggestion
      queryClient.invalidateQueries(ADHERENCE_KEYS.alertsAll(props.studyId), { refetchActive: true, refetchInactive: true})
```

Reply:
@nickgros - you are talking about `{queryKey:..` YES! I didn't catch that and strangely neither did TS. Good call. It is strange that it still seemed to work. I will try the 'reset'